milli/src/update/index_documents/typed_chunk.rs

```diff
@@ -432,40 +684,39 @@ pub(crate) fn write_typed_chunk_into_index(
-                let mut cursor = embeddings.into_cursor()?;
-                while let Some((key, value)) = cursor.move_on_next()? {
-                    let docid = key.try_into().map(DocumentId::from_be_bytes).unwrap();
-                    let data = pod_collect_to_vec(value);
-                    // it is a code error to have embeddings and not expected_dimension
-                    let embeddings =
-                        crate::vector::Embeddings::from_inner(data, expected_dimension)
-                            // code error if we somehow got the wrong dimension
-                            .unwrap();
-
-                    if embeddings.embedding_count() > usize::from(u8::MAX) {
-                        let external_docid = if let Ok(Some(Ok(index))) = index
-                            .external_id_of(wtxn, std::iter::once(docid))
-                            .map(|it| it.into_iter().next())
-                        {
-                            index
-                        } else {
-                            format!("internal docid={docid}")
-                        };
-                        return Err(crate::Error::UserError(crate::UserError::TooManyVectors(
-                            external_docid,
-                            embeddings.embedding_count(),
-                        )));
-                    }
-                    for (embedding, writer) in embeddings.iter().zip(&writers) {
-                        writer.add_item(wtxn, docid, embedding)?;
-                    }
+            let merger = embeddings_builder.build();
+            let mut iter = merger.into_stream_merger_iter()?;
+            while let Some((key, value)) = iter.next()? {
+                let docid = key.try_into().map(DocumentId::from_be_bytes).unwrap();
+                let data = pod_collect_to_vec(value);
+                // it is a code error to have embeddings and not expected_dimension
+                let embeddings = crate::vector::Embeddings::from_inner(data, expected_dimension)
+                    // code error if we somehow got the wrong dimension
+                    .unwrap();
+
+                if embeddings.embedding_count() > usize::from(u8::MAX) {
+                    let external_docid = if let Ok(Some(Ok(index))) = index
+                        .external_id_of(wtxn, std::iter::once(docid))
+                        .map(|it| it.into_iter().next())
+                    {
+                        index
+                    } else {
+                        format!("internal docid={docid}")
+                    };
+                    return Err(crate::Error::UserError(crate::UserError::TooManyVectors(
+                        external_docid,
+                        embeddings.embedding_count(),
+                    )));
+                }
+                for (embedding, writer) in embeddings.iter().zip(&writers) {
+                    writer.add_item(wtxn, docid, embedding)?;
```

Comment:
@dureuill Can you double-check this part, please?

Reply:
sorry I don't think I understand the change, is it related to merging late in the process?

Reply:
Hello @dureuill, sorry, the vector TypedChunk indexing is a pretty big code, I have changed a part of this code to index [several grenad files via a merger](https://github.com/meilisearch/meilisearch/blob/39c83cb3d9ff61367ef7c496061b918c38c2074a/milli/src/update/index_documents/typed_chunk.rs#L628-L651) instead of a unique one. We want to ensure I did everything correctly in that change. I put below the full code:
https://github.com/meilisearch/meilisearch/blob/39c83cb3d9ff61367ef7c496061b918c38c2074a/milli/src/update/index_documents/typed_chunk.rs#L624-L772